### ISSUE TITLE
MuseSounds reverb has flipped polarity in one of its channels

### ISSUE THREAD

**Issue by teetow** [musescore/MuseScore#13993]
**Describe the bug**
In the MuseSounds Strings pack (and possibly others) the reverb tail has one of its channels flipped

**To Reproduce**
1. Make pretty music with MuseSounds
2. Stop playback
3. Listen to the reverb tail
4. Notice one of its channels has its polarity flipped

**Expected behavior**
An actual stereo reverb that doesn't make my ears uncomfortable

**Screenshots**
Waveform while instruments are playing (no issue)
![image](https://user-images.githubusercontent.com/183582/197588718-ec1fa6d3-21d5-478b-b964-83fe79c22e04.png)

Reverb tail immediately after stopping playback (one channel is flipped)
![image](https://user-images.githubusercontent.com/183582/197588890-ad66f528-0a95-4091-bab4-4fbae0ab167b.png)

**Platform information**
 - OS: Windows 10

**Additional context**
Using October 24 nightly

**Comment by teetow** [comment]
Yes, it still happens. I literally just opened up a new project and added some MuseStrings. I also tried with MS Basic, the issue is even more pronounced there since those samples are a lot drier. The reason you might have a harder time reproducing it now is that stopping playback will also kill the verb tail, at least in MuseSounds. Not sure if / when that changed. 

However, the issue is clearly still present. Here's a screenshot from Audition again, with the Phase analysis window showing. As you can see, it's firmly in the red, meaning the current signal is 180° out of phase. The issue is also clearly audible, exactly the kind of hollowed-out disembodied sound you'd expect.

![image](https://user-images.githubusercontent.com/183582/208666156-9a3b1794-076c-4c56-a759-8cfe37d69b8c.png)

**Comment by iamtesch** [comment]
@teetow -- just to clarify, you are experiencing this with MS Basic as well, correct?

**Comment by teetow** [comment]
Yup.

**Comment by teetow** [comment]
In the meantime, is there any chance we can just get a toggle for disabling the realtime reverb entirely? I'd have that as a feature request anyway.

**Comment by RomanPudashkin** [comment]
@teetow hello! Is this still an issue in 4.1.1? Thanks!

**Comment by LWinterberg** [comment]
![image](https://github.com/musescore/MuseScore/assets/87814144/e96c5990-ccda-4947-abe1-6c1b1439ca66)
looks like things are in phase now.

**Comment by teetow** [comment]
As far as I can tell, it's still there. Here are some in-depth steps to reproduce:

1. Create a Musescore project with a series of short notes
2. Render the project twice: once with reverb, once without
3. Add both tracks in parallel to a DAW (e.g. Reaper)
4. Invert polarity of one of the tracks, so that the dry signal is cancelled out
5. Render the result to onlyverb.wav -- you now have a file with *only* the verb tail
6. Analyze the rendered file (onlyverb.wav). It's got significant material to the sides (meaning, it's out of phase).
7. Split the file into L and R channels, invert polarity of one of the channels
8. Analyze the result. It's significantly more in-phase now, and sounds much more pleasing.

So at some stage in your reverb DSP chain (either inside the plugin or outside) a channel gets its polarity flipped. If you can't figure out where, I suggest just adding an extra step in the Reverb plugin that flips one of them again =)

**Comment by sampleeditor** [comment]
@teetow You are 100% correct. The reverb is out of phase. 
This is likely a signal processing issue within Muse Score and/or the Reverb itself.

**Comment by DmitryArefiev** [comment]
@teetow Please check the latest 4.6 nightly build (https://github.com/musescore/MuseScore/actions/runs/17539321607)

> Steps to reproduce the behavior:
> 
> 1. Make pretty music with MuseSounds
> 2. Stop playback
> 3. Listen to the reverb tail
> 4. Notice one of its channels has its polarity flipped

So the issue occurs (noticeable by ear) only after playback is stopped?

**Comment by teetow** [comment]
Still present in the 4.6 nightly build.

The issue is present regardless. Here are phase diagrams of _not_ stopping playback, just previewing a note as normal:

<img width="268" height="457" alt="Image" src="https://github.com/user-attachments/assets/5b7e94b5-9dce-4f7a-985f-feaa2914ff00" />

<img width="268" height="457" alt="Image" src="https://github.com/user-attachments/assets/fe65be48-7d4c-4a77-beac-c9cfd381a7be" />

**Comment by DmitryArefiev** [comment]
Do you mean it goes to 'Inverse' zone at the end?

**Comment by teetow** [comment]
> Do you mean it goes to 'Inverse' zone at the end?

Exactly, as if the reverb has one of its channels flipped. If you bring an exported recording into e.g. Audacity and flip the polarity of just one channel, the note itself sounds bad, but the tail much more natural.

**Comment by DmitryArefiev** [comment]
@teetow Thanks!

@bkunda The problem is in **Reverb** plugin itself. 
* I **can reproduce** it in **Audition** with **MuseFX>Reverb** (on dry signal exported from MSS)

<img width="1452" height="1277" alt="Image" src="https://github.com/user-attachments/assets/c0890729-8761-4592-8a5b-3b9be7ef7009" />


* And **can't** reproduce with NI **Raum** inserted to Aux channel in MSS

**Comment by DmitryArefiev** [comment]
@teetow Can you try this build please? https://github.com/musescore/MuseScore/actions/runs/20464930218

**Comment by DmitryArefiev** [comment]
@sampleeditor Can you try build from https://github.com/musescore/MuseScore/pull/31562 please?
